apps/lib/apps.c

```diff
@@ -1003,10 +1003,16 @@ int load_key_certs_crls(const char *uri, int format, int maybe_stdin,
         ctx = OSSL_STORE_open_ex(uri, libctx, propq, get_ui_method(), &uidata,
                                  params, NULL, NULL);
     }
-    if (ctx == NULL)
+    if (ctx == NULL) {
+        if (!quiet)
+            BIO_printf(bio_err, "Could not open file or uri for loading");
         goto end;
-    if (expect > 0 && !OSSL_STORE_expect(ctx, expect))
+    }
+    if (expect > 0 && !OSSL_STORE_expect(ctx, expect)) {
+        if (!quiet)
+            BIO_printf(bio_err, "Could not find");
```

Comment:
At this point, the `"Could not find"` string seems not the the best/appropriate one.
How about `"Failed to set expectation to find"`?

Reply:
I'm preparing a little PR tweaking and clarifying the code w.r.t. `failed` and `expect`.

A question likely best to @levitte :
Is it advisable to call `OSSL_STORE_expect(ctx, expect)` also in the case that `expect == 0`,
which means in this function body context that more than one type of credentials is requested to get loaded?
Which is something subtly different rom having no expectations (which `man OSSL_STORE_expect` talks about).

Reply:
IMO the _Failed to set expectation to find_ is too cryptic and does not really mean anything to the end user.

Reply:
The "Could not find" would be misleading, namely it would be output in case anything goes wrong calling `OSSL_STORE_expect()`.
Which is quite different from the expected type of credential not being present/found in the given store!

Reply:
Since you consider "Failed to set expectation to find <xyz>" too cryptic,
how about "Internal error trying to load"?

Reply:
It's not *wrong* to call `OSSL_STORE_expect()` with 0 (it's even documented!).  Not sure if that actually matters very much in practice, but it does depend on the backend, so again, not wrong.

Reply:
Hmm, reading the doc I was already expecting that it should not be wrong to call `OSSL_STORE_expect(ctx, 0)`,
but my question was: is it advisable? In particular if we cannot assume anything about the fetched loaders?
Can it bring any benefit, and OTOH does it bring risks of needless failure?

So, would you recommend changing
`if (expect > 0 && !OSSL_STORE_expect(ctx, expect))` to
`if (expect >= 0 && !OSSL_STORE_expect(ctx, expect))` ?

Reply:
And can I conclude from your response that you see no difference between having no expectations on the type(s) of credentials to load and expecting to load more than one type of credentials (which is the case here)?